sched/scheduler/task_runner.go

```diff
@@ -29,37 +31,73 @@ func runTaskAndLog(
 	task sched.TaskDefinition,
 	markCompleteOnFailure bool,
 	stat stats.StatsReceiver) error {
-	// Log StartTask Message to SagaLog
+
+	// Log the StartTask message with an empty payload. This will be overwritten with more data when we have it.
 	err := saga.StartTask(taskId, nil)
 	if err != nil {
 		return err
 	}
-	// runtask on worker
-	processStatus, err := worker.RunAndWait(task)
-	if err != nil {
-		stat.Counter("failedTaskRunCounter").Inc(1)
 
-		// if we should mark tasks complete even if an error occurs set processStatus
-		// and don't return.  Otherwise return the error and the task will
-		// get rescheduled.
-		if markCompleteOnFailure {
-			processStatus.Error = err.Error()
-			processStatus.ExitCode = DeadLetterExitCode
-			log.Printf(
-				`Error Running Task %v: dead lettering task after max retries.  
+	// Handle errors the same way for Start()/Wait().
+	// Overwrites selected processStatus fields and leaves other values intact.
+	// If err, check markCompletedOnFailure. If true, log and continue. If false, return err immediately.
+	processStatus := runner.ProcessStatus{}
+	workerErrStatus := func(err error) error {
+		if err != nil {
+			stat.Counter("failedTaskRunCounter").Inc(1)
+
+			// if we should mark tasks complete even if an error occurs set processStatus
+			// and don't return.  Otherwise return the error and the task will
+			// get rescheduled.
+			if markCompleteOnFailure {
+				processStatus.Error = err.Error()
+				processStatus.ExitCode = DeadLetterExitCode
```

Comment:
why is this called 'DeadLetterExitCode'?  Suggest 'FailureExitCode' or 'CompleteWithFailure'